src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -326,6 +326,7 @@ class RsErrorAnnotator : Annotator, HighlightRangeExtension {
     }
 
     private fun RsRefLikeType.needsLifetime(): Boolean {
+        if (mul != null) return false
```

Comment:
Based on discussion in https://github.com/intellij-rust/intellij-rust/commit/02c242ce0653609eb0c271993dbd5f69e25f1f46#commitcomment-20830146, I think that perhaps instead of `mul != null` check, you should check if perhaps the owning `RsTypeReference` type is `RustReferenceType`?